tests/py_test.py

```diff
@@ -545,10 +584,13 @@ def asserter(stmt):
         asserter(size == len(buf[offset:])-4)
         buf, offset = util.RemoveSizePrefix(buf, offset)
     if file_identifier:
+        # test again after removal of size_prefix
+        asserter(util.GetBufferIdentifier(buf, offset) == file_identifier)
+        asserter(util.BufferHasIdentifier(buf, offset, file_identifier=file_identifier))
         asserter(MyGame.Example.Monster.Monster.MonsterBufferHasIdentifier(buf, offset))
-    else:
```

Comment:
Why was this branch of the conditional removed?

Reply:
It was invalid. It assumed that the file identifier would not be present if there was no file identifier. However, theoretically, the vtable offset could be the file identifier, although this is wildly unlikely(maybe impossible) without some kind of intentionall medling

Reply:
@pyottamus It seems to me that it would be more useful, instead of calling it "invalid", to make a comment explaining this rare, possibly impossible, caveat...